Corona-Warn-App/src/main/java/de/rki/coronawarnapp/nearby/ExposureStateUpdateWorker.kt

```diff
@@ -7,39 +7,33 @@ import com.google.android.gms.common.api.ApiException
 import com.google.android.gms.nearby.exposurenotification.ExposureNotificationClient
 import de.rki.coronawarnapp.exception.ExceptionCategory
 import de.rki.coronawarnapp.exception.NoTokenException
-import de.rki.coronawarnapp.exception.TransactionException
 import de.rki.coronawarnapp.exception.reporting.report
+import de.rki.coronawarnapp.risk.RiskLevelTask
 import de.rki.coronawarnapp.storage.ExposureSummaryRepository
-import de.rki.coronawarnapp.transaction.RiskLevelTransaction
+import de.rki.coronawarnapp.task.common.DefaultTaskRequest
+import de.rki.coronawarnapp.util.di.AppInjector
 import timber.log.Timber
 
 class ExposureStateUpdateWorker(val context: Context, workerParams: WorkerParameters) :
     CoroutineWorker(context, workerParams) {
-    companion object {
-        private val TAG = ExposureStateUpdateWorker::class.simpleName
-    }
 
     override suspend fun doWork(): Result {
         try {
             Timber.v("worker to persist exposure summary started")
             val token = inputData.getString(ExposureNotificationClient.EXTRA_TOKEN)
                 ?: throw NoTokenException(IllegalArgumentException("no token was found in the intent"))
-
             Timber.v("valid token $token retrieved")
-
-            val exposureSummary = InternalExposureNotificationClient
-                .asyncGetExposureSummary(token)
-
-            ExposureSummaryRepository.getExposureSummaryRepository()
-                .insertExposureSummaryEntity(exposureSummary)
-            Timber.v("exposure summary state updated: $exposureSummary")
-
-            RiskLevelTransaction.start()
+            InternalExposureNotificationClient
+                .asyncGetExposureSummary(token).also {
+                    ExposureSummaryRepository.getExposureSummaryRepository()
+                        .insertExposureSummaryEntity(it)
+                    Timber.v("exposure summary state updated: $it")
+                }
+
+            AppInjector.component.taskController.submit(DefaultTaskRequest(RiskLevelTask::class))
```

Comment:
Couldn't we constructor inject this?

Reply:
Can't yet, will make this possible soon.

Reply:
Will be made possible by #1503